VisualPinball.Unity/VisualPinball.Unity/Game/CameraClipPlane.cs

```diff
@@ -0,0 +1,142 @@
+// Visual Pinball Engine
+// Copyright (C) 2021 freezy and VPE Team
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+using System; 
+using Unity.Mathematics;
+using UnityEngine;
+
+
+namespace VisualPinball.Unity
+{
+	[ExecuteAlways]
+	public class CameraClipPlane : MonoBehaviour
+	{
+		
+		[NonSerialized]
+		public Camera Camera;
+
+		//Stores the table bounds. 
+		private Bounds tableBounds;
+		private Vector3 previousCameraPosition = Vector3.zero;
+
+		private void OnEnable()
+		{
+			SetClipPlanes(0.001f, 9f); 
+		}
+
+		private void Update()
+		{
+			if(!GetCameraComponent())
+			{
+				return;
+			}
+
+			if(previousCameraPosition != Camera.transform.position)
+			{
+				UpdateClipPlanes();
+				previousCameraPosition = Camera.transform.position; 
+			}
+		}
+
+		/// <summary>
+		/// Updates the camera clipping planes based on the table bounds.
+		/// </summary>
+		public bool UpdateClipPlanes()
```

Comment:
Return value isn't used.

Reply:
It isnt used at the moment, return value provided in case other camera controllers that people use or make need to know if it was successfully set since it can fail. Not essential by any means though.